tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -862,6 +894,29 @@ where
             )?;
         }
 
+        let line_number = if self.display_line_number { meta.line() } else { None };
+
+        if self.display_filename {
+            if let Some(filename) = meta.file() {
+                write!(
+                    writer,
+                    "{}{}{}",
+                    dimmed.paint(filename),
+                    dimmed.paint(":"),
+                    if line_number.is_some() { "" } else { " " }
+                )?;
+            }
+        }
+
+        if let Some(line_number) = line_number {
+            write!(
+                writer,
+                "{}{} ",
+                dimmed.paint(line_number.to_string()),
+                dimmed.paint(":")
+            )?;
+        }
```

Comment:
As I mentioned [here](https://github.com/tokio-rs/tracing/issues/1326#issuecomment-986284256), I'm still unconvinced that we should display line numbers if both targets _and_ file names are disabled...a line number like ':156' doesn't make a lot of sense if we don't know what file or module it's in. But, trying to handle that case nicely will complicate the implementation significantly. 

@davidbarsky what do you think? Is it worthwhile to make sure we do the "right thing" if a user calls `with_line_number(true)` but never enables file names? What if they've also called `.with_target(false)`?

Reply:
> @davidbarsky what do you think? Is it worthwhile to make sure we do the "right thing" if a user calls `.with_line_number(true)` but never enables file names? 

Just saw this. While I think that it's worth doing the right thing, I'd like to consider redesigning the configuration interface of `fmt` holistically. To that end, I'd rather accept this PR-as-is designed, and fix it later. 

At the very least, the lone line number will confusing, but it won't cause runtime panics like other misconfigurations can.

Reply:
To be clear, I'm not entirely sure what the holistic redesign should look like, but naively, I might want to try bundling lines and file formatting information into a single struct where a missing target/file is _very_ obvious to the end usr.